### PR TITLE
Autocomplete - update selectedItem option if new list value is equal to input text (T991350)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets.editors/autocomplete.tests.js [hunk 1]
```diff
@@ -1469,6 +1469,49 @@ QUnit.module('regressions', {
...
+        this.instance.option({
+            searchTimeout: timeToWait,
+            onSelectionChanged: selectionChangedStub
+        });
+
+        this.keyboard
+            .type('item 2');
+
+        this.clock.tick(timeToWait + 100);
+        const $listItems = $(this.instance.content()).find(`.${LIST_ITEM_CLASS}`);
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/autocomplete.tests.js]:
```suggestion
        const $listItems = this.instance.$content().find(`.${LIST_ITEM_CLASS}`);
```

File: testing/tests/DevExpress.ui.widgets.editors/autocomplete.tests.js [hunk 2]
```diff
@@ -1469,6 +1469,49 @@ QUnit.module('regressions', {
         assert.strictEqual(selectionChangedStub.lastCall.args[0].selectedItem, null);
     });
 
+    QUnit.test('onSelectionChanged event should trigger on item selection if its text is equal to input text (T991350)', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/autocomplete.tests.js]: Let's add a new submodule with instance configuration in `beforeEach` to minimize copypaste?

File: testing/tests/DevExpress.ui.widgets.editors/autocomplete.tests.js (hunk 1 above)
Comment (alexander-kotov-dx) [reply testing/tests/DevExpress.ui.widgets.editors/autocomplete.tests.js]: we do not have $content() for Autocomplete

**Review by Zedwag — APPROVED** [review]

File: testing/tests/DevExpress.ui.widgets.editors/autocomplete.tests.js [hunk 3]
```diff
@@ -1469,6 +1469,45 @@ QUnit.module('regressions', {
...
+                onSelectionChanged: this.selectionChangedStub
+            });
+        }
+    }, () => {
+        QUnit.test('onSelectionChanged event should trigger on item selection if its text is equal to input text (T991350)', function(assert) {
+            this.keyboard
+                .type('item 2');
+
+            this.clock.tick(this.timeToWait + 100);
+            const $listItems = $(this.instance.content()).find(`.${LIST_ITEM_CLASS}`);
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/autocomplete.tests.js:1488]:
```suggestion
            const $listItems = this.instance.content().find(`.${LIST_ITEM_CLASS}`);
```

We'are refactoring all such accessors in a parallel branch now

**Review by ksercs — APPROVED** [review]

**Pull request merged**